frontend/src/components/App.tsx

```diff
@@ -31,11 +39,45 @@ function App() {
               element={<AdminQuizEditForm />}
             />
             <Route path="/admin/users" element={<AdminUser />} />
+            {/*#endregion*/}
 
-            <Route path="/signin" element={<UserSignIn />} />
-            <Route path="/signup" element={<UserSignUp />} />
-            <Route path="/profile" element={<UserProfile />} />
-            <Route path="/quizzes" element={<UserQuizzes />} />
+            {/*#region Auth*/}
+            <Route
+              path="/signin"
+              element={
+                <RequireGuest>
+                  <UserSignIn />
+                </RequireGuest>
+              }
```

Comment:
It is better to create a component for middleware and pass the component/element to be protected, so that when adding a new user role you just need to create a new middleware and pass the component/element.

1. Create a new file for middleware 
2. Apply the logic in the middleware
3. Import the middle ware in App.tsx or where your routes is
4. Pass the component/element that you want to protect
```js
<UserRoute path="/quiz/:id" exact component={Quiz} />
<AdminRoute path="/quiz/create/" exact component={QuizCreate} />
```
> Assuming we have a `Quiz` and `QuizCreate` component

Reply:
https://gist.github.com/mjackson/d54b40a094277b7afdd6b81f51a0393f#route-composition-in-react-router-v6

henlo, basically for v6, custom route classes are heavily discouraged and this is sort of the recommended way

Reply:
ill just refactor a bit to reduce loc

Reply:
@ejanton-sun 
I agree with the article. For me since it is explained as good practice (which also makes sense) you can follow this approach <Route element={ <MuStBeAuTh element={ <A/> }> }>. EJ’s comment is also not wrong since it gives easier way to reuse route logic.

As conclusion, you can follow the new practice since it solves some props issues.

Reply:
@jstephend-sun @jeremiahC duly noted, since v6 is recently been released(stable version)